griptape/tasks/prompt_task.py

```diff
@@ -107,8 +107,7 @@ def after_run(self) -> None:
         structure = self.structure
         conversation_memory = self.conversation_memory
         if (
-            structure is not None
-            and structure.conversation_memory_strategy == "per_task"
+            (structure is None or (structure is not None and structure.conversation_memory_strategy == "per_task"))
```

Comment:
Should be equivalent to:
```py
 (structure is None or structure.conversation_memory_strategy == "per_task")
 ```

If `structure is None` is true, then the rhs of the `or` won't even be checked.

Reply:
Great call